### ISSUE TITLE
Caps Lock LED lights up when using remote desktop

### ISSUE THREAD

**Issue by Snaptags** [MaxGyver83/neo2-llkh#38]
After replacing the native driver by the hook, I had to do the same thing on the machine I'm working on via remote desktop. Works nicely, but whenever I press Mod3 the Caps Lock LED is toggled, first press ON, second press OFF.

That's bad because on the Host Windows this actually activates Caps Lock :(

**Comment by Snaptags** [comment]
The best solution were if I didn't have to install anything on the Client PC at all. But I guess the keyboard hook is not THAT low-level…

**Comment by MaxGyver83** [comment]
> After replacing the native driver by the hook, I had to do the same thing on the machine I'm working on via remote desktop. Works nicely, but whenever I press Mod3 the Caps Lock LED is toggled, first press ON, second press OFF.
> 
> That's bad because on the Host Windows this actually activates Caps Lock :(

I see the same behavior when I run `neo-llkh.exe` on Windows in VirtualBox with Linux as host.

> The best solution were if I didn't have to install anything on the Client PC at all. But I guess the keyboard hook is not THAT low-level…

When I change the keyboard layout on the (Linux) host via `setxkbmap`, I still have QWERTZ in Windows. When I change the keyboard layout via [KMonad](https://github.com/david-janssen/kmonad) on the host, it's also applied in Windows. _KMonad_ is also available for Windows but it still has some issues. Maybe it matches your needs better than _neo-llkh_.

**Comment by Snaptags** [comment]
Any idea if there is an easy fix for the Caps Lock thing? _neo-llkh_ is perfect otherwise.

**Comment by MaxGyver83** [comment]
I'll try to find out!

**Comment by Snaptags** [comment]
Let me know if I can help debugging etc. I'll try KMonad, too, just found your neo.kd for it. Thanks for providing the config!

**Comment by Snaptags** [comment]
KMonad does not look that promising unfortunately. It has a quite high input lag and most of my keys don't work at all, e.g. SHIFT or CTRL…

**Comment by MaxGyver83** [comment]
> KMonad does not look that promising unfortunately. It has a quite high input lag and most of my keys don't work at all, e.g. SHIFT or CTRL…

Maybe your config doesn't use the setting `fallthrough true` in `defcfg`? Without this setting all keys not defined in `defsrc` will be "eaten" by KMonad.

**Comment by MaxGyver83** [comment]
I made some tests with VirtualBox and noticed that CapsLock gets synched between host and guest system even without `neo-llkh.exe`. So (at least for VirtualBox) it's not a bug in `neo-llkh.exe`. (Btw, I found a feature request for VirtualBox: https://www.virtualbox.org/ticket/14053.)

How does the CapsLock key behave in your remote desktop session when you don't use `neo-llkh`?

**Comment by MaxGyver83** [comment]
At first, I thought that it could be possible to somehow catch the CapsLock key, use it as Neo's Mod3 and then make it disappear. But I think it's the other way round: The host system processes the CapsLock key first and toggles the state and the LED on the host system, then it's being passed to the remote/guest system and for `neo-llkh` it's too late to prevent a capslock state switch.

**Comment by Snaptags** [comment]
Without `neo-llkh` on host and the remote machine, the CapsLock state ist synced between the machines. Turning it on on either machine keeps it on everywhere until I turn it off again anywhere.

**Comment by Snaptags** [comment]
Did lots of research yesterday on that. Syncing CapsLock state seems to be an essential feature of RDP. Tried lots of registry hacks to disable it, nothing worked…

BUT: I just fixed my problem using a screwdriver and changed the CAPS LOCK key mapping to # by flashing a new firmware… Now I have two "right Mod3" keys and no caps lock key at all. Problem solved 😄 

Now I'm perfectly happy, because your solution is WAY better than the "official" setup. That never worked in exotic apps (e.g. like GIMP). This is no problem using neo-llkh.
